### DISCUSSION TITLE
grid grid-cols-[20%, 40%, 40%]

### DISCUSSION THREAD

**Discussion by kishore-uc** [tailwindlabs/tailwindcss#13539] [Help]
I want custom grid layout with 3 different grid col sizes, so I made 3col grid with 20%, 40%, 40%.    but it did not work, any reason why it not working in tailwind.
I am using angular flowbite and tailwind. 
<div class="grid grid-cols-[20%, 40%, 40%] gap-4">
  <div class="p-4">First Item (20%)</div>
  <div class="p-4">Second Item (40%)</div>
  <div class=" p-4">Third Item (40%)</div>
</div>

![grid2](https://github.com/tailwindlabs/tailwindcss/assets/45749990/c332451b-1789-467b-b795-41d5ea04747c)

In "https://play.tailwindcss.com/" also I have added the above html grid structure it also gives same
![grid-play-tailwind](https://github.com/tailwindlabs/tailwindcss/assets/45749990/3b2955a2-8740-4d2f-9ec8-6ef6090e78d8)

**Comment by wongjn** [comment]
You'd have the column track size values of `grid-cols-` separated by [spaces escaped as underscores](https://tailwindcss.com/docs/content#using-spaces-and-underscores) like `grid-cols-[20%_40%_40%]`.

**Reply by kishore-uc** [reply]
It works well, Thank you so much @wongjn for your response.
same way, How to make 12 grid system in tailwind like bootstrap
eg of bootstrap
<div class="container text-center">
  <div class="row">
    <div class="col-2">
      text
    </div>
    <div class="col-5">
      text
    </div>
    <div class="col-5">
     text
    </div>
  </div>
</div>
but col need to be exactly divided
container is 1440px
first col is 354px
second col is 614px
third col is 472px  some padding will be there between 3 cols ignore it for now.

**Reply by wongjn** [reply]
You could consider using a CSS grid with a custom `grid-cols-*`, similar to what you have tried in your original question, along with applying other CSS properties to achieve your desired layout.

**Reply by kishore-uc** [reply]
thank you for your reply,  
you mean like this ?

![Uploading grid3.png…]()

<div class="grid grid-cols-[2.5, 5.5, 6] gap-4">
  <div class="p-4">First Item </div>
  <div class="p-4">Second Item</div>
  <div class=" p-4">Third Item</div>
</div>, I tried it wont work, and also we need exact size of the columns, which will help us a lot,

**Reply by wongjn** [reply]
Not like that – you need to use HTML to apply Tailwind classes like:
```html
<div class="…">
  <div class="…">First Item</div>
  <div class="…">Second Item</div>
  <div class="…">Third Item</div>
</div>
```
For exact column sizes, you could apply CSS properties that give you the exact sizings.
